source/npm/util.js

```diff
@@ -107,3 +109,74 @@ exports.checkIgnoreStrategy = ({files}) => {
 		`);
 	}
 };
+
+// New files added part of .npmignore or not part of files-attribute in "package.json"
+exports.checkNewFiles = async (newFiles, filesFromPackageDotJson) => {
+	const rootDir = pkgDir.sync();
+	const npmignoreExists = fs.existsSync(path.resolve(rootDir, '.npmignore'));
+
+	let result = [];
+
+	if (filesFromPackageDotJson) {
+		result = getFilesNotPartOfFileAttribute(filesFromPackageDotJson, newFiles);
+	}
+
+	if (npmignoreExists) {
+		result = result.concat(await getFilesIgnoredByDotnpmignore(newFiles));
+	}
+
+	return result;
+};
+
+async function getFilesIgnoredByDotnpmignore(fileList) {
+	if (!Array.isArray(fileList)) {
+		throw new TypeError('expected array, but got {typeof fileList}');
+	}
+
+	const result = [];
+	await ignoreWalker({
+		path: pkgDir.sync(),
+		ignoreFiles: ['.npmignore']
+	})
+		.then(WhiteList => {
+			if (!Array.isArray(WhiteList)) {
+				throw new TypeError(`Expected array, but got ${typeof WhiteList} `);
+			}
+
+			// Every file in fileList should be part of the WhiteList
+			fileList.forEach(item => {
+				const found = WhiteList.find(WhiteListItem => {
+					return WhiteListItem === item;
+				});
+				if (found === undefined) {
+					result.push(item);
+				}
+			});
+		});
+
+	return result;
+}
+
+function getFilesNotPartOfFileAttribute(filesFromPackageDotJson, fileList) {
+	if (!Array.isArray(fileList)) {
+		throw new TypeError(`expected array, but got ${typeof fileList}`);
+	}
+
+	const result = [];
+	const globMatcher = new Minimatch(getGlobPattern(filesFromPackageDotJson),
+		{matchBase: true});
+	fileList.forEach(item => {
+		if (!globMatcher.match(item)) {
+			result.push(item);
+		}
+	});
+	return result;
```

Comment:
Either use a `for-of` or a `.filter`,  which is more appropriate here.

https://github.com/isaacs/minimatch#minimatchfilterpattern-options

Reply:
I used the the ```minimatch.filter```-function and negated the glob-pattern with an leading ```!```.